### ISSUE TITLE
Processing dbms.jvm.additional broken since 4.4.9

### ISSUE THREAD

**Issue by nioertel** [neo4j/neo4j#12914]
## Bug report

When upgrading to Neo4j 4.4.9 we realized that processing of the property `dbms.jvm.additional` is not working anymore for JVM parameters that contain spaces.

Extract from neo4j.conf:
`dbms.jvm.additional="-XX:+HeapDumpOnOutOfMemoryError" "-XX:HeapDumpPath=/tmp/" "-XX:OnOutOfMemoryError=/usr/bin/env handle-oom.sh %p /tmp/java_pid%p.hprof" "-Dlog4j.configurationFile=classpath:log4j2.json"`

Output of `cat /proc/$PID/cmdline | tr '\000' '\n'`
```
/opt/java/openjdk/bin/java
-cp
/var/lib/neo4j/plugins/*:/var/lib/neo4j/conf/*:/var/lib/neo4j/lib/*
-Xms4194304k
-Xmx4194304k
-XX:+HeapDumpOnOutOfMemoryError
-XX:HeapDumpPath=/tmp/
"-XX:OnOutOfMemoryError=/usr/bin/env
handle-oom.sh
%p
/tmp/java_pid%p.hprof"
-Dlog4j.configurationFile=classpath:log4j2.json
-Dfile.encoding=UTF-8
com.neo4j.server.enterprise.EnterpriseEntryPoint
--home-dir=/var/lib/neo4j
--config-dir=/var/lib/neo4j/conf
```

You can see that the OnOutOfMemoryError parameter has been incorrectly split into 4 separate parameters.
Up until Neo4j 4.4.8 this issue was not present.

In case the format in which parameters with spaces have to be provided within `dbms.jvm.additional` changed, please refer me to the documentation.

**Neo4j Version:** 4.4.9
**Operating System:** RHEL 7.9 w/ Docker 20.10.5
**API:** Docker

### Steps to reproduce
Try to start a Neo4j docker container with the parameters provided above:
```
docker run --env=NEO4J_ACCEPT_LICENSE_AGREEMENT=yes --env=NEO4J_dbms_jvm_additional='"-XX:+HeapDumpOnOutOfMemoryError" "-XX:HeapDumpPath=/tmp/" "-XX:OnOutOfMemoryError=/usr/bin/env handle-oom.sh %p /tmp/java_pid%p.hprof" "-Dlog4j.configurationFile=classpath:log4j2.json"' neo4j:4.4.9-enterprise
```

### Expected behavior
Neo4j should start up without errors showing an output like this:
```
2022-08-03 13:01:24.576+0000 INFO  Starting...
2022-08-03 13:01:25.310+0000 INFO  This instance is ServerId{7d2fcb67} (7d2fcb67-eb01-4ad8-b906-e5dad1949c8f)
2022-08-03 13:01:27.847+0000 INFO  ======== Neo4j 4.4.9 ========
2022-08-03 13:01:31.714+0000 INFO  Sending metrics to CSV file at /var/lib/neo4j/metrics
2022-08-03 13:01:31.786+0000 INFO  Bolt enabled on 0.0.0.0:7687.
2022-08-03 13:01:33.369+0000 INFO  Remote interface available at http://localhost:7474/
2022-08-03 13:01:33.377+0000 INFO  id: 873067D4D1C8CD6469576A7FDCAC8B881C3EABFE7919C3FBCB14F2536E6BDD72
2022-08-03 13:01:33.378+0000 INFO  name: system
2022-08-03 13:01:33.378+0000 INFO  creationDate: 2022-08-03T13:01:28.93Z
2022-08-03 13:01:33.379+0000 INFO  Started.
```

### Actual behavior
Neo4j fails to start with the following output:
```
Error: Could not find or load main class "-XX:OnOutOfMemoryError=.usr.bin.env
Caused by: java.lang.ClassNotFoundException: "-XX:OnOutOfMemoryError=.usr.bin.env
```

**Comment by nioertel** [comment]
Probably related to this commit: neo4j/docker-neo4j@17c0acc598fab6765a5366164b056524eeaf8647
I will also raise a bug in the neo4j-docker repository.

**Comment by klaren** [comment]
This should be fixed in 4.4.11, https://neo4j.com/release-notes/database/neo4j-4-4-11/ `Jvm additional setting should not split inside quoted values`. That in tandem with https://github.com/neo4j/docker-neo4j/pull/376 

Can you verify this?

**Comment by nioertel** [comment]
Confirmed. Issue can be closed.
